### PR TITLE
build: Enable JIT and optimize the WAMR for envoy

### PR DESCRIPTION
**Pull request by leyao-daily**
JIT Mode can be open now. Finish the optimization of WAMR for Envoy

Co-authored-by: Liang He <liang.he@intel.com>
Signed-off-by: Le Yao <le.yao@intel.com>

Commit Message: Enable JIT and optimize the WAMR for envoy
Additional Description: JIT Mode can be open now. Finish the optimization of WAMR for Envoy
Risk Level: Low
Testing: WIP
Docs Changes: N/A
Release Notes: N/A
Platform Specific Features: N/A
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by junr03** [comment]
Hi @leyao-daily I noticed this PR is in Draft. Would you like me to assign a maintainer to start the code review process, or wait until you mark ready for review?

/wait-any

**Comment by leyao-daily** [comment]
> Hi @leyao-daily I noticed this PR is in Draft. Would you like me to assign a maintainer to start the code review process, or wait until you mark ready for review?
> 
> /wait-any

I still work on some local tests. I think it can be reviewed after I finish testing and mark it ready for review. Thanks

**Comment by alyssawilk** [comment]
Just to be aware, we've been making some improvements to Envoy tooling for more prompt reviews, but draft PRs are not included in the tooling.  I'm updating the contribution guidelines to make that clear (https://github.com/envoyproxy/envoy/pull/17063/files) , but as this PR predates the repokitteh warning I just want to call out that draft PRs are likely not going to get fast turnaround. Please consider marking this as ready for review if it stalls, and you want the assignee to take a look!

File: bazel/repository_locations.bzl
Comment (PiotrSikora) [inline bazel/repository_locations.bzl:712]: Could you update it to a more recent commit? It looks that there were some relevent bugfixes since then.

Comment (leyao-daily) [reply bazel/repository_locations.bzl:712]: Okay

**Comment by leyao-daily** [comment]
I think it's ready to merge. Thanks :)

**Comment by leyao-daily** [comment]
@PiotrSikora Thanks for your comments. I have update to the latest commit. And can you help review it. LOL Thanks. :)

**Review by lizan — APPROVED** [review]

**Pull request merged**
